src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,40 @@ pub trait Iterator {
         None
     }
 
+    /// Creates an iterator starting at the same point, but stepping by
+    /// the given amount at each iteration.
+    ///
+    /// Note that it will always return the first element of the range,
+    /// regardless of the step given.
+    ///
+    /// # Panics
+    ///
+    /// If the given step is `0`, the method will panic if debug assertions are
+    /// enabled.
```

Comment:
I just looked at how the other methods in `Iterator` that panic were phrased and tried doing something similar for consistency. 

I just checked now, and I didn't know of `debug_assert!`, which is what the other methods are using. I've been mislead! :P I'll fix it soon. I also have to figure out what to write on the Unstable Book to make that tidy check pass 🙄